### PR TITLE
Feat/challenge 03 loops conditionals

### PR DESCRIPTION
**Pull request by Manel0712**

### CODE REVIEW

File: challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_01.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Create a program that starts with a number variable.
+    The program will return the variable is even or odd.
+
+    Change the variable to test it works for different cases.
+    Print result.
+
+    HINT: modulus operator ( % ) is your friend
+*/
+
+let number = 9;
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_01.js]: ⚠️ FIX: use 🇬🇧, please.

⚠️ FIX:  change the variable to test it works for different cases. 

⚠️ FIX: add a comment to print expected output.

File: challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_02.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+} else if (age >= 13 && age < 18) {
+  console.log("Nice, you're a teenager!");
+} else if (age >= 18) {
+  console.log("Most places consider you an adult.");
+} else {
+  console.log("What? How did this happen!?");
+}
+
+/*
+  The error is that the conditions only check that the age is greater than the number indicated, but not that they are less than the number indicated in the following condition.
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_02.js:46]: How would you fix it?

⚠️ FIX: redefine script to fix the bug

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/*
+    1. Create a for loop that will print out all the even numbers between 10 and 40.
+    
+    2. Create a for loop that will print out all the odd numbers between 10 and 40.
+*/
+for (let i = 10; i <= 40; i = i + 2) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js]: Variables are always defined using camelCase convention, they never start by capital letter.
You should also take in account what they are storing.
⚠️ FIX: rename properly for variables in both cases

These loops should iterate over all numbers in between 10 and 40, printing only the odd/even ones, depending on its own logic.
⚠️ FIX: iterate over all numbers and print only the odd/even ones

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_01.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/*
+    1. Create a for loop that loops while index is less or equal than 10.
+    2. Each loop:
+        * create a new random number between 1 and 10 (hint: check documentation regarding Math.floor and Math.random methods)
+        * evaluate if random number is greater than 5
+        * if it is equal or greater than 5 print "Logged In!"
+        * if it is less than 5 print "Logged Out!"
+*/
+
+for (let i = 0; i <= 10; i++) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_01.js]: ⚠️ FIX: rename properly for variables

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    Half way there!
+    Checkpoint! 60
+    Checkpoint! 70
+    Checkpoint! 80
+    Checkpoint! 90
+    You made it!
+    All, done!
+*/
+
+for (let i = 1; i <= 100; i++) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js]: This is the output of your script
```
'Half way there'

'You made it!'

'Checkpoint! 10'
'Checkpoint! 20'
'Checkpoint! 30'
'Checkpoint! 40'
'¡All, done!'
'Checkpoint! 60'
'Checkpoint! 70'
'Checkpoint! 80'
'Checkpoint! 90'
```

have you tested it before pushing? 🤔

⚠️ FIX: needs work.

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_00.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    1. Create a variable called "loggedIn" that starts off as false.
+    2. Create a while loop that loops while "loggedIn" is false.
+    3. Each loop, print out: "Incorrect login credentials"
+    3. After 3 loop iterations, change "loggedIn" to be true.
+    4. After the loops is done print out: "Sucessfully logged in!"
+
+    HINT1: Be careful for an infinite loop!
+*/
+
+let loggedln = false;
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_00.js:11]: ⚠️ FIX: rename properly variable `i`

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_01.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+/*
+    1. Create a while loop that will print out all the even numbers between 10 and 40.
+
+    2. Create a while loop that will print out all the odd numbers between 10 and 40.
+*/
+
+let i = 10;
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_01.js]: ⚠️ FIX: rename properly variables `i` and `c`

These loops should iterate over all numbers in between 10 and 40, printing only the odd/even ones, depending on its own logic.
⚠️ FIX: iterate over all numbers and print only the odd/even ones

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    Checkpoint! 30
+    Checkpoint! 40
+    Half way there!
+    Checkpoint! 60
+    Checkpoint! 70
+    Checkpoint! 80
+    Checkpoint! 90
+    You made it!
+    All, done!
+*/
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js:20]: This is the output of your script
```
'Half way there!'

'You made it!'

'Checkpoint! 10'
'Checkpoint! 20'
'Checkpoint! 30'
'Checkpoint! 40'
100
'All, done!'
'Checkpoint! 60'
'Checkpoint! 70'
'Checkpoint! 80'
'Checkpoint! 90'
```

have you tested it before pushing? 🤔

⚠️ FIX: needs work.

**Review by tictools — CHANGES_REQUESTED** [review]

**Review by Enma1920 — APPROVED** [review]

**Review by Enma1920 — APPROVED** [review]

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+/*
+    1. Create a for loop that will print out all the even numbers between 10 and 40.
+    
+    2. Create a for loop that will print out all the odd numbers between 10 and 40.
+*/
+for (let evennumbers = 10; evennumbers <= 40; evennumbers++) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js]: @Manel0712  @Enma1920 
Pay atention! Variables are defined using camelCase convention
```suggestion
for (let evenNumbers = 10; evenNumbers <= 40; evenNumbers++) {
```

FIX: apply suggestion

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  if (evennumbers % 2 === 0) {
+    console.log(evennumbers);
+  }
+}
+for (let oddnumbers = 10; oddnumbers <= 40; oddnumbers++) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js]: Also here. 

FIX: use camelCase convetion

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js]: FIX: name properly variables

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  i % 10 === 0
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js]: It's not a good idea use ternaries nested inside ternaries. Avoid this kind of implementations, they can be so difficult to read.

FIX: take a look at soltuions and refactor ternary

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_01.js [hunk 13]
```diff
@@ -0,0 +1,20 @@
+/*
+    1. Create a while loop that will print out all the even numbers between 10 and 40.
+
+    2. Create a while loop that will print out all the odd numbers between 10 and 40.
+*/
+
+let evennumbers = 10;
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_01.js]: FIX: Also here.

FIX: use camelCase convetion in all cases

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+let i = 1;
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js]: FIX: name properly variables

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+while (i <= 100) {
+  i % 10 === 0
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js]: It's not a good idea use ternaries nested inside ternaries. Avoid this kind of implementations, they can be so difficult to read.

FIX: take a look at soltuions and refactor ternary

**Review by tictools — CHANGES_REQUESTED** [review]

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+    Checkpoint! 40
+    Half way there!
+    Checkpoint! 60
+    Checkpoint! 70
+    Checkpoint! 80
+    Checkpoint! 90
+    You made it!
+    All, done!
+*/
+function readnumbers(number) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js]: CamelCase

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+    Half way there!
+    Checkpoint! 60
+    Checkpoint! 70
+    Checkpoint! 80
+    Checkpoint! 90
+    You made it!
+    All, done!
+*/
+let numbers = 1;
+function readnumbers(number) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js]: CamelCase

**Review by Enma1920 — APPROVED** [review]

**Review by Enma1920 — APPROVED** [review]

**Review by tictools — CHANGES_REQUESTED** [review]

**Review by Enma1920 — APPROVED** [review]

**Review by tictools — APPROVED** [review]

**Pull request merged**